homeassistant/components/influxdb.py

```diff
@@ -222,91 +224,75 @@ def influx_event_listener(event):
 
         json_body[0]['tags'].update(tags)
 
-        _write_data(json_body)
+        nonlocal write_error
+        for retry in range(max_tries+1):
+            try:
+                influx.write_points(json_body)
+                if write_error:
+                    _LOGGER.error("Resumed writing to InfluxDB")
+                    write_error = False
+                break
+            except (exceptions.InfluxDBClientError, IOError):
+                if retry == max_tries:
+                    if not write_error:
+                        _LOGGER.error("Error writing to InfluxDB")
+                        write_error = True
+                else:
+                    time.sleep(RETRY_DELAY)
```

Comment:
Can we expose this new block with retry into thread class? Yes, the sleep is okay while you run your own thread, but for the code understanding, it would there is no sleep inside `setup()`

Reply:
Yes, I made that change and it is better 👍 even if the nesting of the loop is a bit deep now.